src/deepqmc/app.py

```diff
@@ -31,22 +32,29 @@
 )
 
 
+def nuclear_configurations(coords, unit):
+    unit_factor = {'bohr': 1.0, 'angstrom': 1 / 0.52917721092}[unit]
+    coords = jax.numpy.asarray(coords)
+    coords = coords if coords.ndim == 3 else [coords]
+    return [unit_factor * jax.numpy.asarray(coord) for coord in coords]
+
+
 def instantiate_ansatz(hamil, ansatz):
     import haiku as hk
 
     return hk.without_apply_rng(
         hk.transform_with_state(
-            lambda r, return_mos=False: ansatz(hamil)(r, return_mos)
+            lambda phys_conf, return_mos=False: ansatz(hamil)(phys_conf, return_mos)
         )
     )
 
 
-def train_from_factories(hamil, ansatz, sampler, **kwargs):
+def train_from_factories(hamil, ansatz, sampler, nuc_configs, **kwargs):
     from .sampling import chain
     from .train import train
 
     ansatz = instantiate_ansatz(hamil, ansatz)
-    sampler = chain(*sampler[:-1], sampler[-1](hamil))
+    sampler = [chain(*sampler[:-1], sampler[-1](hamil, R)) for R in nuc_configs]
```

Comment:
Is this a list over the entire dataset of nuclear configs, or over the current training batch?

Reply:
This is a list over the entire dataset. For large datasets we'll need to be more efficient, e.g. initializing samplers on the fly.